filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/batch/GcsBatchIoCommand.scala

```diff
@@ -114,7 +114,34 @@ case class GcsBatchDeleteCommand(
   */
 sealed trait GcsBatchGetCommand[T] extends SingleFileGcsBatchIoCommand[T, StorageObject] {
   def file: GcsPath
-  private val blob = file.blob
+  private val blob = {
+    val result = file.blob
+    if (Option(result.getName).forall(_.trim.nonEmpty)) {
```

Comment:
I think `forall` is true if the option is empty. Is that right? Did you mean `exists`?